src/poll.rs

```diff
@@ -175,6 +175,229 @@ libc_bitflags! {
     }
 }
 
+/// Timeout argument for [`poll`].
+#[derive(Debug, Clone, Copy, Eq, PartialEq, Ord, PartialOrd)]
+pub struct PollTimeout(i32);
+
+impl PollTimeout {
+    /// Blocks indefinitely.
+    ///
+    /// > Specifying a negative value in timeout means an infinite timeout.
+    pub const NONE: Self = Self(-1);
+    /// Returns immediately.
+    ///
+    /// > Specifying  a timeout of zero causes poll() to return immediately, even if no file
+    /// > descriptors are ready.
+    pub const ZERO: Self = Self(0);
+    /// Blocks for at most [`std::i32::MAX`] milliseconds.
+    pub const MAX: Self = Self(i32::MAX);
+    /// Returns if `self` equals [`PollTimeout::NONE`].
+    pub fn is_none(&self) -> bool {
+        // > Specifying a negative value in timeout means an infinite timeout.
+        *self <= Self::NONE
+    }
```

Comment:
I think this should remain `<=` as this is the safer option. The speedup gained from `==` over `<=` will be a couple nanoseconds at most. I can see a possible bug in the future arising otherwise.

Reply:
> I can see a possible bug in the future arising otherwise.

Would like to elaborate on it a bit? Maybe give an example?

Reply:
If in the future for some reason on some specific platform other non-negative values are supported because it is useful or necessary, this would then cause this function to be incorrect if not also changed. By sticking close to the documentation it provides the safest approach considering cross-platform and future proofing.

Reply:
Ah, I see, makes sense to me, would you like to add a comment here stating why we are doing it in this way in case someone will accidentally change it in the future